src/main/java/ai/knowly/langtorch/store/vectordb/PGVectorStore.java

```diff
@@ -0,0 +1,342 @@
+package ai.knowly.langtorch.store.vectordb;
+
+import ai.knowly.langtorch.processor.EmbeddingProcessor;
+import ai.knowly.langtorch.schema.embeddings.EmbeddingInput;
+import ai.knowly.langtorch.schema.embeddings.EmbeddingOutput;
+import ai.knowly.langtorch.schema.io.DomainDocument;
+import ai.knowly.langtorch.schema.io.Metadata;
+import ai.knowly.langtorch.store.vectordb.integration.VectorStore;
+import ai.knowly.langtorch.store.vectordb.integration.pgvector.PGVectorService;
+import ai.knowly.langtorch.store.vectordb.integration.pgvector.SqlCommandProvider;
+import ai.knowly.langtorch.store.vectordb.integration.pgvector.schema.PGVectorQueryParameters;
+import ai.knowly.langtorch.store.vectordb.integration.pgvector.schema.PGVectorStoreSpec;
+import ai.knowly.langtorch.store.vectordb.integration.pgvector.schema.PGVectorValues;
+import ai.knowly.langtorch.store.vectordb.integration.pgvector.schema.distance.DistanceStrategy;
+import ai.knowly.langtorch.store.vectordb.integration.schema.SimilaritySearchQuery;
+import com.google.common.flogger.FluentLogger;
+import com.google.common.primitives.Floats;
+import com.google.inject.Inject;
+import com.pgvector.PGvector;
+import lombok.NonNull;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+
+/**
+ * A vector store implementation using PostgreSQL and PGVector for storing and querying vectors.
+ */
+public class PGVectorStore implements VectorStore {
+
+    private static final int EMBEDDINGS_COLUMN_COUNT = 2;
+    private static final int EMBEDDINGS_INDEX_ID = 0;
+    private static final int EMBEDDINGS_INDEX_VECTOR = 1;
+    private static final int METADATA_COLUMN_COUNT = 4;
+    private static final int METADATA_INDEX_ID = 0;
+    private static final int METADATA_INDEX_KEY = 1;
+    private static final int METADATA_INDEX_VALUE = 2;
+    private static final int METADATA_INDEX_VECTOR_ID = 3;
+    private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+    @NonNull
+    private final EmbeddingProcessor embeddingsProcessor;
+    private final PGVectorStoreSpec pgVectorStoreSpec;
+    private final SqlCommandProvider sqlCommandProvider;
+    @NonNull
+    private final PGVectorService pgVectorService;
+    private final DistanceStrategy distanceStrategy;
+
+    @Inject
+    public PGVectorStore(@NonNull EmbeddingProcessor embeddingsProcessor,
+                         PGVectorStoreSpec pgVectorStoreSpec,
+                         @NonNull PGVectorService pgVectorService,
+                         DistanceStrategy distanceStrategy,
+                         boolean shouldOverWriteDatabase) throws SQLException {
```

Comment:
I think this only applies to table where we create necessary tables if not there. We should put this into pgVectorStoreSpec and rename to overwriteExistingTables?